spring-2013/hw1/src/CreditCardValidator.hs

```diff
@@ -0,0 +1,22 @@
+module CreditCardValidator
+    ( toDigits
+    , doubleEveryOther
+    , sumDigits
+    , validate
+    ) where
+
+toDigits :: Integer -> [Integer]
+toDigits n
+    | n <= 0 = []
+    | otherwise = map readChar . show $ n
+    where readChar = read . (:[])
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther = reverse . zipWith (*) (cycle [1, 2]) . reverse
+
+sumDigits :: [Integer] -> Integer
+sumDigits = sum . (toDigits =<<)
```

Comment:
I agree avoiding the fanciness is better. As a rule, if my function is list-only, I use list-specific constructs. That's because the ratio of confusion/usefulness is not very good for the list monad (IMO).